libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/kotlin/adapter/VariablesAdapter.kt

```diff
@@ -60,25 +49,25 @@ private fun List<NamedType>.writeToResponseCodeBlock(context: KotlinContext): Co
   return builder.build()
 }
 
-private fun NamedType.writeToResponseCodeBlock(context: KotlinContext): CodeBlock {
+private fun IrVariable.writeToResponseCodeBlock(context: KotlinContext): CodeBlock {
   val adapterInitializer = context.resolver.adapterInitializer(type, false, context.jsExport, customScalarAdapters)
   val builder = CodeBlock.builder()
-  val propertyName = context.layout.propertyName(graphQlName)
+  val propertyName = context.layout.propertyName(name)
 
   if (type.isOptional()) {
     builder.beginControlFlow("if ($value.%N is %T)", propertyName, KotlinSymbols.Present)
   }
-  builder.addStatement("$writer.name(%S)", graphQlName)
+  builder.addStatement("$writer.name(%S)", name)
   builder.addSerializeStatement(adapterInitializer, propertyName)
   if (type.isOptional()) {
     builder.endControlFlow()
-    if (defaultValue is IrBooleanValue) {
-      builder.beginControlFlow("else if ($withBooleanDefaultValues)")
-      builder.addStatement("$writer.name(%S)", graphQlName)
+    if (defaultValue != null) {
+      builder.beginControlFlow("else if ($withDefaultValues)")
+      builder.addStatement("$writer.name(%S)", name)
       builder.addStatement(
           "%M.toJson($writer, $customScalarAdapters, %L)",
-          KotlinSymbols.BooleanAdapter,
-          defaultValue.value,
+          KotlinSymbols.NullableAnyAdapter,
+          defaultValue.codeBlock(),
```

Comment:
This is the most important change. `serializeVariables` now serializes **all** the variables if asked to (instead of just the boolean variables before)